src/compiler/scala/tools/nsc/backend/jvm/ClassfileWriters.scala

```diff
@@ -257,12 +257,12 @@ abstract class ClassfileWriters {
       }
     }
 
-    private final class WithStatsWriter(underlying: ClassfileWriter) extends ClassfileWriter {
+    private final class WithStatsWriter(statistics: Statistics with Global#GlobalStats, underlying: ClassfileWriter)
+      extends ClassfileWriter {
       override def write(className: InternalName, bytes: Array[Byte], paths: CompilationUnitPaths): Unit = {
-        val stats = frontendAccess.unsafeStatistics
-        val snap = stats.startTimer(stats.bcodeWriteTimer)
+        val snap = statistics.startTimer(statistics.bcodeWriteTimer)
         underlying.write(className, bytes, paths)
-        stats.stopTimer(stats.bcodeWriteTimer, snap)
+        statistics.stopTimer(statistics.bcodeWriteTimer, snap)
```

Comment:
I'd prefer exposing statistics through `frontendAccess`, as drafted here https://github.com/scala/scala/commit/e1fb0b825882b4bf2b19f4f9f7c23f6c28c03ee7#diff-0102bb4f2a2247733fb4f45a6417a485R248 - or is there a good reason against?

Reply:
ping here